File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        dict = {}

        for x in nums:
            count = dict.get(x)
            if count:
                count+=1
                dict.update({x:count})
            else:
                dict.update({x:1})

        return len(dict)!=len(nums)

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hash = new HashMap<>();
        for(int i=0; i<nums.length;i++) {
            hash.put(nums[i],i);
        }
        
        for(int i=0; i<nums.length;i++) {
            int search = target-nums[i];
            if(hash.containsKey(search) && hash.get(search) != i) {
                return new int[]{i,hash.get(search)};
            }
        }
        return null;
    }
}
